Measurement/twinkles_light_curves.ipynb

```diff
@@ -0,0 +1,480 @@
+    "plt.scatter(x_image_2, y_image_2, c='g', marker='+', label='Lensed AGN Image 2', s=400, lw=2, zorder=10)\n",
+    "for idx in range(len(src_pix)):\n",
+    "    src_loc = src_pix[idx]\n",
+    "    plt.scatter(src_loc[0], src_loc[1], c='orange', s=100, zorder=1)\n",
+    "plt.legend(fontsize=32)\n",
+    "plt.xlabel('X Pixels', size=28)\n",
+    "plt.ylabel('Y Pixels', size=28)\n",
+    "plt.xticks(size=18)\n",
+    "plt.yticks(size=18)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We see that the AGN image #2 is identified as its own source very close to the true position but that the lensing galaxy and AGN image #1 are blended together and identified as a single source in the output catalog. Let's make this easy on ourselves and focus on building a lightcurve for AGN image #2 below which should be well identified in all 10 r-band images available.\n",
+    "\n",
+    "To do catalog matching we will use Astropy's `match_to_catalog_sky` method. This will require creating SkyCoord objects with the ra, dec coordinates of our truth catalog in the next cell and then in each visit doing the same with the source catalogs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "lens_coords = SkyCoord(test_lens['ra']*u.deg, test_lens['dec']*u.deg)\n",
+    "image_coords = SkyCoord(test_images['ra']*u.deg, test_images['dec']*u.deg)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we will loop through all the r-band visits and get three things from the butler:\n",
+    "\n",
+    "1) Source catalogs: These will have all the sources in each visit. We will use this to match to our truth catalog and get fluxes.\n",
+    "\n",
+    "2) Calexps: The processed images. We will need this to get the information necessary to convert the source catalog fluxes into magntiudes with a `photoCalib`. When we have a `photoCalib` object we can use `instFluxToMagnitude` to convert `base_PsfFlux_instFlux` values in the source catalog to magnitudes.\n",
+    "\n",
+    "3) Calexp Metadata: The metadata will contain information from the FITS headers of the images and we will use this to get the MJD of each visit for our light curve.\n",
+    "\n",
+    "Once we have all this information for each visit we use `match_to_catalog_sky` with our truth locations and identify the source catalog entry that is our AGN image."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "image_mags = []\n",
+    "image_errs_high = []\n",
+    "image_errs_low = []\n",
+    "image_times = []\n",
+    "\n",
+    "for dataId in r_band_subset.cache:\n",
+    "    \n",
+    "    # Get data products from Butler\n",
+    "    src_cat = butler.get('src', dataId = dataId).asAstropy().to_pandas()\n",
+    "    calexp = butler.get('calexp', dataId = dataId)\n",
```

Comment:
You can, instead, use `calexp_photoCalib` instead of getting the `calexp` and then getting the `photoCalib`.